### PR TITLE
fix(rnn): bounds check seq_len_max in BlockLSTM ops to prevent OOB access

### LINKED ISSUE #124974: tf.raw_ops.BlockLSTMGrad crashes with fatal check failure (SIGABRT) in Tensor::Slice when seq_len_max exceeds tensor dimension size
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

2.22.0-dev20260809

### Custom code

Yes

### OS platform and distribution

WINDOWS 10 

### Mobile device

_No response_

### Python version

_No response_

### Bazel version

_No response_

### GCC/compiler version

_No response_

### CUDA/cuDNN version

_No response_

### GPU model and memory

_No response_

### Current behavior?

When calling tf.compat.v1.raw_ops.BlockLSTMGrad with a seq_len_max value that is greater than the time/sequence dimension of the input tensor x, an unhandled slicing bounds error occurs inside BlockLSTMGradOp<>::Compute().

When the C++ kernel calls tensorflow::Tensor::Slice(), a hard check failure is triggered:
Check failed: limit <= dim0_size (10 vs. 2)

Because this validation isn't handled gracefully before the slicing operation, the process terminates abruptly with a core dump (SIGABRT). TensorFlow should validate seq_len_max against the leading dimension of x during kernel execution and throw a tf.errors.InvalidArgumentError.

### Standalone code to reproduce the issue

```shell
import tensorflow as tf

# seq_len_max is 10, but leading dimension of x is 2 or smaller in slice logic
seq_len_max = 10

x = tf.zeros([10, 2, 16], dtype=tf.half)
cs_prev = tf.zeros([2, 0], dtype=tf.half)
h_prev = tf.zeros([2, 0], dtype=tf.half)
w = tf.zeros([16, 0], dtype=tf.half)
wci = tf.zeros([5], dtype=tf.half)
wcf = tf.zeros([16], dtype=tf.half)
wco = tf.zeros([13], dtype=tf.half)
b = tf.zeros([0], dtype=tf.half)

i = tf.zeros([10, 2, 0], dtype=tf.half)
cs = tf.zeros([10, 2, 0], dtype=tf.half)
f = tf.zeros([10, 2, 0], dtype=tf.half)
o = tf.zeros([10, 2, 0], dtype=tf.half)
ci = tf.zeros([10, 2, 0], dtype=tf.half)
co = tf.zeros([10, 2, 0], dtype=tf.half)
h = tf.zeros([10, 2, 0], dtype=tf.half)
cs_grad = tf.zeros([2, 0], dtype=tf.half)
h_grad = tf.zeros([2, 0], dtype=tf.half)

# Triggers fatal assertion in BlockLSTMGradOp::Compute -> Tensor::Slice
res = tf.compat.v1.raw_ops.BlockLSTMGrad(
    seq_len_max=seq_len_max,
    x=x,
    cs_prev=cs_prev,
    h_prev=h_prev,
    w=w,
    wci=wci,
    wcf=wcf,
    wco=wco,
    b=b,
    i=i,
    cs=cs,
    f=f,
    o=o,
    ci=ci,
    co=co,
    h=h,
    cs_grad=cs_grad,
    h_grad=h_grad,
    use_peephole=False
)
```

### Relevant log output

```shell
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1786343711.008116     353 port.cc:153] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
I0000 00:00:1786343711.010427     353 cudart_stub.cc:31] Could not find cuda drivers on your machine, GPU will not be used.
I0000 00:00:1786343711.091406     353 cpu_feature_guard.cc:227] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F AVX512_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1786343714.024784     353 port.cc:153] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
I0000 00:00:1786343714.028355     353 cudart_stub.cc:31] Could not find cuda drivers on your machine, GPU will not be used.
E0000 00:00:1786343714.762301     353 cuda_platform.cc:57] failed call to cuInit: INTERNAL: CUDA error: Failed call to cuInit: UNKNOWN ERROR (303)
=== Source Location Trace: ===
external/xla/xla/stream_executor/cuda/cuda_status.cc:50

/mnt/d/tf-env/lib/python3.12/site-packages/tensorflow/python/ops/math_ops.py:1157: RuntimeWarning: overflow encountered in cast
  min_limit = np_dtype(np.maximum(in_dtype.min, out_real_dtype.min))
E0000 00:00:1786343714.897236     353 util.cc:131] oneDNN supports DT_HALF only on platforms with AVX-512. Falling back to the default Eigen-based implementation if present.
F0000 00:00:1786343714.915105     353 tensor.cc:1195] Check failed: limit <= dim0_size (10 vs. 2)
[symbolize_elf.inc : 356] RAW: Unable to get high fd: rc=0, limit=1024
*** Check failure stack trace: ***
    @     0x7f42230d5a54  absl::lts_20260526::log_internal::LogMessage::SendToLog()
    @     0x7f42230d59d6  absl::lts_20260526::log_internal::LogMessage::Flush()
    @     0x7f4222c9affa  tensorflow::Tensor::Slice()
    @     0x7f41f916b05d  tensorflow::(anonymous namespace)::SliceHelper<>::UnalignedSlice()
    @     0x7f41f916aa91  tensorflow::(anonymous namespace)::SliceHelper<>::InputSlice()
    @     0x7f41f9180f52  tensorflow::BlockLSTMGradOp<>::Compute()
    @     0x7f421dd84e90  tensorflow::ThreadPoolDevice::Compute()
    @     0x7f421dd33b96  tensorflow::(anonymous namespace)::SingleThreadedExecutorImpl::Run()
    @     0x7f421dd00f2d  tensorflow::FunctionLibraryRuntimeImpl::RunSync()
    @     0x7f421dd0c516  tensorflow::ProcessFunctionLibraryRuntime::RunMultiDeviceSync()
    @     0x7f421dd12590  tensorflow::ProcessFunctionLibraryRuntime::RunSync()
    @     0x7f42013fc4cd  tensorflow::KernelAndDeviceFunc::Run()
    @     0x7f42013aacba  tensorflow::EagerKernelExecute()
    @     0x7f42013b486e  tensorflow::ExecuteNode::Run()
    @     0x7f4201531756  tensorflow::EagerExecutor::SyncExecute()
    @     0x7f42013aa661  tensorflow::(anonymous namespace)::EagerLocalExecute()
    @     0x7f42013a7db0  tensorflow::DoEagerExecute()
    @     0x7f42013ab81b  tensorflow::EagerExecute()
    @     0x7f4200f85c47  tensorflow::EagerOperation::Execute()
    @     0x7f42013f8e47  tensorflow::CustomDeviceOpHandler::Execute()
    @     0x7f4200afcee5  TFE_Execute
    @     0x7f41e03b5213  TFE_Py_FastPathExecute_C()
    @     0x7f41c936a813  pybind11::detail::argument_loader<>::call<>()
    @     0x7f41c936a74f  pybind11::cpp_function::initialize<>()::{lambda()#1}::__invoke()
    @     0x7f41c933f599  pybind11::cpp_function::dispatcher()
    @           0x5821ef  (unknown)
    @           0x5492b5  _PyObject_MakeTpCall
    @           0x5d7389  _PyEval_EvalFrameDefault
    @           0x5d58bb  PyEval_EvalCode
    @           0x608b52  (unknown)
    @           0x6b5263  (unknown)
    @           0x6b4fca  _PyRun_SimpleFileObject
    @           0x6b4dff  _PyRun_AnyFileObject
    @           0x6bce65  Py_RunMain
    @           0x6bc94d  Py_BytesMain
    @     0x7f422835a1ca  (unknown)
    @     0x7f422835a28b  __libc_start_main
    @           0x657b05  _start
Aborted (core dumped)
```

### PR DESCRIPTION
**Pull request by veerjain-1**
Resolves #124974. This PR introduces bounds checking for the `seq_len_max` parameter in the core TensorFlow LSTM and LSTM gradient operations (`BlockLSTMOp` and `BlockLSTMGradOp`). Previously, an oversized or negative `seq_len_max` bypassed validation and triggered a fatal abort (SIGABRT) in `Tensor::Slice`. The parameter is now validated to ensure it falls within the bounds of `[0, timelen]`.

### CODE REVIEW

**Review by dmiltr3 — CHANGES_REQUESTED** [review]
Please add unit tests covering `seq_len_max` bounds validation for both `BlockLSTM` and `BlockLSTMGrad` in `tensorflow/python/kernel_tests/nn_ops/rnn_cell_test.py` to ensure invalid bounds (`seq_len_max < 0` and `seq_len_max > timelen`) raise `InvalidArgumentError` and valid boundary values (`0`, `timelen`) execute as expected.

Suggested tests to add to `rnn_cell_test.py`:

```python
  @test_util.run_in_graph_and_eager_modes
  def testBlockLSTMSeqLenMaxBounds(self):
    timelen, batch_size, input_size, cell_size = 4, 2, 3, 5
    x = constant_op.constant(0.1, shape=[timelen, batch_size, input_size], dtype=dtypes.float32)
    cs_prev = constant_op.constant(0.1, shape=[batch_size, cell_size], dtype=dtypes.float32)
    h_prev = constant_op.constant(0.1, shape=[batch_size, cell_size], dtype=dtypes.float32)
    w = constant_op.constant(0.1, shape=[input_size + cell_size, 4 * cell_size], dtype=dtypes.float32)
    wci = constant_op.constant(0.1, shape=[cell_size], dtype=dtypes.float32)
    wcf = constant_op.constant(0.1, shape=[cell_size], dtype=dtypes.float32)
    wco = constant_op.constant(0.1, shape=[cell_size], dtype=dtypes.float32)
    b = constant_op.constant(0.1, shape=[4 * cell_size], dtype=dtypes.float32)

    # Valid boundary cases: seq_len_max == 0 and seq_len_max == timelen
    for valid_seq_len_max in [0, 2, timelen]:
      res = self.evaluate(
          gen_rnn_ops.BlockLSTM(
              seq_len_max=constant_op.constant(valid_seq_len_max, dtype=dtypes.int64),
              x=x,
              cs_prev=cs_prev,
              h_prev=h_prev,
              w=w,
              wci=wci,
              wcf=wcf,
              wco=wco,
              b=b,
              forget_bias=0.0,
              cell_clip=-1.0,
              use_peephole=False,
          )
      )
      self.assertEqual(len(res), 7)

    # Invalid cases: seq_len_max < 0 or seq_len_max > timelen
    for invalid_seq_len_max in [-1, timelen + 1, 100]:
      with self.assertRaisesRegex(
          (ValueError, errors_impl.InvalidArgumentError),
          r"seq_len_max must be between 0 and",
      ):
        self.evaluate(
            gen_rnn_ops.BlockLSTM(
                seq_len_max=constant_op.constant(invalid_seq_len_max, dtype=dtypes.int64),
                x=x,
                cs_prev=cs_prev,
                h_prev=h_prev,
                w=w,
                wci=wci,
                wcf=wcf,
                wco=wco,
                b=b,
                forget_bias=0.0,
                cell_clip=-1.0,
                use_peephole=False,
            )
        )

  @test_util.run_in_graph_and_eager_modes
  def testBlockLSTMGradSeqLenMaxBounds(self):
    timelen, batch_size, input_size, cell_size = 4, 2, 3, 5
    x = constant_op.constant(0.1, shape=[timelen, batch_size, input_size], dtype=dtypes.float32)
    cs_prev = constant_op.constant(0.1, shape=[batch_size, cell_size], dtype=dtypes.float32)
    h_prev = constant_op.constant(0.1, shape=[batch_size, cell_size], dtype=dtypes.float32)
    w = constant_op.constant(0.1, shape=[input_size + cell_size, 4 * cell_size], dtype=dtypes.float32)
    wci = constant_op.constant(0.1, shape=[cell_size], dtype=dtypes.float32)
    wcf = constant_op.constant(0.1, shape=[cell_size], dtype=dtypes.float32)
    wco = constant_op.constant(0.1, shape=[cell_size], dtype=dtypes.float32)
    b = constant_op.constant(0.1, shape=[4 * cell_size], dtype=dtypes.float32)
    i = constant_op.constant(0.1, shape=[timelen, batch_size, cell_size], dtype=dtypes.float32)
    cs = constant_op.constant(0.1, shape=[timelen, batch_size, cell_size], dtype=dtypes.float32)
    f = constant_op.constant(0.1, shape=[timelen, batch_size, cell_size], dtype=dtypes.float32)
    o = constant_op.constant(0.1, shape=[timelen, batch_size, cell_size], dtype=dtypes.float32)
    ci = constant_op.constant(0.1, shape=[timelen, batch_size, cell_size], dtype=dtypes.float32)
    co = constant_op.constant(0.1, shape=[timelen, batch_size, cell_size], dtype=dtypes.float32)
    h = constant_op.constant(0.1, shape=[timelen, batch_size, cell_size], dtype=dtypes.float32)
    cs_grad = constant_op.constant(0.1, shape=[timelen, batch_size, cell_size], dtype=dtypes.float32)
    h_grad = constant_op.constant(0.1, shape=[timelen, batch_size, cell_size], dtype=dtypes.float32)

    # Valid boundary cases: seq_len_max == 0 and seq_len_max == timelen
    for valid_seq_len_max in [0, 2, timelen]:
      res = self.evaluate(
          gen_rnn_ops.BlockLSTMGrad(
              seq_len_max=constant_op.constant(valid_seq_len_max, dtype=dtypes.int64),
              x=x,
              cs_prev=cs_prev,
              h_prev=h_prev,
              w=w,
              wci=wci,
              wcf=wcf,
              wco=wco,
              b=b,
              i=i,
              cs=cs,
              f=f,
              o=o,
              ci=ci,
              co=co,
              h=h,
              cs_grad=cs_grad,
              h_grad=h_grad,
              use_peephole=False,
          )
      )
      self.assertEqual(len(res), 8)

    # Invalid cases: seq_len_max < 0 or seq_len_max > timelen
    for invalid_seq_len_max in [-1, timelen + 1, 100]:
      with self.assertRaisesRegex(
          (ValueError, errors_impl.InvalidArgumentError),
          r"seq_len_max must be between 0 and",
      ):
        self.evaluate(
            gen_rnn_ops.BlockLSTMGrad(
                seq_len_max=constant_op.constant(invalid_seq_len_max, dtype=dtypes.int64),
                x=x,
                cs_prev=cs_prev,
                h_prev=h_prev,
                w=w,
                wci=wci,
                wcf=wcf,
                wco=wco,
                b=b,
                i=i,
                cs=cs,
                f=f,
                o=o,
                ci=ci,
                co=co,
                h=h,
                cs_grad=cs_grad,
                h_grad=h_grad,
                use_peephole=False,
            )
        )
```

**Comment by veerjain-1** [comment]
Hi team! Just a gentle ping on this PR. Let me know if there's anything else you need from my side or if any changes are required. Thanks!

**Review by dmiltr3 — CHANGES_REQUESTED** [review]
Please add unit tests covering `seq_len_max` bounds validation for both `BlockLSTM` and `BlockLSTMGrad` in `tensorflow/python/kernel_tests/nn_ops/rnn_cell_test.py` to ensure invalid bounds (`seq_len_max < 0` and `seq_len_max > timelen`) raise `InvalidArgumentError` and valid boundary values (`0`, `timelen`) execute as expected.

Suggested tests to add to `rnn_cell_test.py`:

```python
  @test_util.run_in_graph_and_eager_modes
  def testBlockLSTMSeqLenMaxBounds(self):
    timelen, batch_size, input_size, cell_size = 4, 2, 3, 5
    x = constant_op.constant(0.1, shape=[timelen, batch_size, input_size], dtype=dtypes.float32)
    cs_prev = constant_op.constant(0.1, shape=[batch_size, cell_size], dtype=dtypes.float32)
    h_prev = constant_op.constant(0.1, shape=[batch_size, cell_size], dtype=dtypes.float32)
    w = constant_op.constant(0.1, shape=[input_size + cell_size, 4 * cell_size], dtype=dtypes.float32)
    wci = constant_op.constant(0.1, shape=[cell_size], dtype=dtypes.float32)
    wcf = constant_op.constant(0.1, shape=[cell_size], dtype=dtypes.float32)
    wco = constant_op.constant(0.1, shape=[cell_size], dtype=dtypes.float32)
    b = constant_op.constant(0.1, shape=[4 * cell_size], dtype=dtypes.float32)

    # Valid boundary cases: seq_len_max == 0 and seq_len_max == timelen
    for valid_seq_len_max in [0, 2, timelen]:
      res = self.evaluate(
          gen_rnn_ops.BlockLSTM(
              seq_len_max=constant_op.constant(valid_seq_len_max, dtype=dtypes.int64),
              x=x,
              cs_prev=cs_prev,
              h_prev=h_prev,
              w=w,
              wci=wci,
              wcf=wcf,
              wco=wco,
              b=b,
              forget_bias=0.0,
              cell_clip=-1.0,
              use_peephole=False,
          )
      )
      self.assertEqual(len(res), 7)

    # Invalid cases: seq_len_max < 0 or seq_len_max > timelen
    for invalid_seq_len_max in [-1, timelen + 1, 100]:
      with self.assertRaisesRegex(
          (ValueError, errors_impl.InvalidArgumentError),
          r"seq_len_max must be between 0 and",
      ):
        self.evaluate(
            gen_rnn_ops.BlockLSTM(
                seq_len_max=constant_op.constant(invalid_seq_len_max, dtype=dtypes.int64),
                x=x,
                cs_prev=cs_prev,
                h_prev=h_prev,
                w=w,
                wci=wci,
                wcf=wcf,
                wco=wco,
                b=b,
                forget_bias=0.0,
                cell_clip=-1.0,
                use_peephole=False,
            )
        )

  @test_util.run_in_graph_and_eager_modes
  def testBlockLSTMGradSeqLenMaxBounds(self):
    timelen, batch_size, input_size, cell_size = 4, 2, 3, 5
    x = constant_op.constant(0.1, shape=[timelen, batch_size, input_size], dtype=dtypes.float32)
    cs_prev = constant_op.constant(0.1, shape=[batch_size, cell_size], dtype=dtypes.float32)
    h_prev = constant_op.constant(0.1, shape=[batch_size, cell_size], dtype=dtypes.float32)
    w = constant_op.constant(0.1, shape=[input_size + cell_size, 4 * cell_size], dtype=dtypes.float32)
    wci = constant_op.constant(0.1, shape=[cell_size], dtype=dtypes.float32)
    wcf = constant_op.constant(0.1, shape=[cell_size], dtype=dtypes.float32)
    wco = constant_op.constant(0.1, shape=[cell_size], dtype=dtypes.float32)
    b = constant_op.constant(0.1, shape=[4 * cell_size], dtype=dtypes.float32)
    i = constant_op.constant(0.1, shape=[timelen, batch_size, cell_size], dtype=dtypes.float32)
    cs = constant_op.constant(0.1, shape=[timelen, batch_size, cell_size], dtype=dtypes.float32)
    f = constant_op.constant(0.1, shape=[timelen, batch_size, cell_size], dtype=dtypes.float32)
    o = constant_op.constant(0.1, shape=[timelen, batch_size, cell_size], dtype=dtypes.float32)
    ci = constant_op.constant(0.1, shape=[timelen, batch_size, cell_size], dtype=dtypes.float32)
    co = constant_op.constant(0.1, shape=[timelen, batch_size, cell_size], dtype=dtypes.float32)
    h = constant_op.constant(0.1, shape=[timelen, batch_size, cell_size], dtype=dtypes.float32)
    cs_grad = constant_op.constant(0.1, shape=[timelen, batch_size, cell_size], dtype=dtypes.float32)
    h_grad = constant_op.constant(0.1, shape=[timelen, batch_size, cell_size], dtype=dtypes.float32)

    # Valid boundary cases: seq_len_max == 0 and seq_len_max == timelen
    for valid_seq_len_max in [0, 2, timelen]:
      res = self.evaluate(
          gen_rnn_ops.BlockLSTMGrad(
              seq_len_max=constant_op.constant(valid_seq_len_max, dtype=dtypes.int64),
              x=x,
              cs_prev=cs_prev,
              h_prev=h_prev,
              w=w,
              wci=wci,
              wcf=wcf,
              wco=wco,
              b=b,
              i=i,
              cs=cs,
              f=f,
              o=o,
              ci=ci,
              co=co,
              h=h,
              cs_grad=cs_grad,
              h_grad=h_grad,
              use_peephole=False,
          )
      )
      self.assertEqual(len(res), 8)

    # Invalid cases: seq_len_max < 0 or seq_len_max > timelen
    for invalid_seq_len_max in [-1, timelen + 1, 100]:
      with self.assertRaisesRegex(
          (ValueError, errors_impl.InvalidArgumentError),
          r"seq_len_max must be between 0 and",
      ):
        self.evaluate(
            gen_rnn_ops.BlockLSTMGrad(
                seq_len_max=constant_op.constant(invalid_seq_len_max, dtype=dtypes.int64),
                x=x,
                cs_prev=cs_prev,
                h_prev=h_prev,
                w=w,
                wci=wci,
                wcf=wcf,
                wco=wco,
                b=b,
                i=i,
                cs=cs,
                f=f,
                o=o,
                ci=ci,
                co=co,
                h=h,
                cs_grad=cs_grad,
                h_grad=h_grad,
                use_peephole=False,
            )
        )
```

**Comment by veerjain-1** [comment]
Hi @dmiltr3, I have added the requested unit tests covering `seq_len_max` bounds validation in `rnn_cell_test.py` and resolved the merge conflicts. Could you please take another look? Thanks!

**Review by dmiltr3 — CHANGES_REQUESTED** [review]
### CI Failure Investigation & Required Test Fix

The GitHub CI checks failed across multiple platforms:
* [build-linux-x86-cpu](https://github.com/tensorflow/tensorflow/actions/runs/32667670569/job/97264183061)
* [build-linux-x86-cuda](https://github.com/tensorflow/tensorflow/actions/runs/32667670569/job/97264183227)
* [build-linux-x86-cuda13](https://github.com/tensorflow/tensorflow/actions/runs/32667670569/job/97264183168)
* [build-windows-x86](https://github.com/tensorflow/tensorflow/actions/runs/32667670569/job/97264183187)

#### Root Cause
In `tensorflow/core/kernels/rnn/lstm_ops.cc`, consolidating the bounds check error message into `"seq_len_max must be between 0 and ..."` caused 4 existing unit tests in `tensorflow/python/ops/rnn_grad_test.py` to fail (`testBlockLSTMSeqLenMaxNegative`, `testBlockLSTMSeqLenMaxTooLarge`, `testBlockLSTMGradSeqLenMaxNegative`, `testBlockLSTMGradSeqLenMaxTooLarge`) because their assertion regex expects `"seq_len_max must be <= timelen"` or `"seq_len_max must be >= 0"`.

#### Proposed Fix
Please update the expected error message regex in `tensorflow/python/ops/rnn_grad_test.py`:

```diff
diff --git a/tensorflow/python/ops/rnn_grad_test.py b/tensorflow/python/ops/rnn_grad_test.py
index a5d3e09..b61c94d 100644
--- a/tensorflow/python/ops/rnn_grad_test.py
+++ b/tensorflow/python/ops/rnn_grad_test.py
@@ -120,7 +120,7 @@ class RNNGradTest(test.TestCase):
   def testBlockLSTMSeqLenMaxTooLarge(self):
     w, b, x, cs_prev, h_prev, w_peephole = self._block_lstm_inputs()
     with self.assertRaisesRegex(
-        errors_impl.InvalidArgumentError, "seq_len_max must be <= timelen"
+        errors_impl.InvalidArgumentError, r"seq_len_max must be between 0 and"
     ):
       self.evaluate(
           self._block_lstm(w, b, x, cs_prev, h_prev, w_peephole, seq_len_max=10)
@@ -128,7 +128,7 @@ class RNNGradTest(test.TestCase):
   def testBlockLSTMSeqLenMaxNegative(self):
     w, b, x, cs_prev, h_prev, w_peephole = self._block_lstm_inputs()
     with self.assertRaisesRegex(
-        errors_impl.InvalidArgumentError, "seq_len_max must be >= 0"
+        errors_impl.InvalidArgumentError, r"seq_len_max must be between 0 and"
     ):
       self.evaluate(
           self._block_lstm(w, b, x, cs_prev, h_prev, w_peephole, seq_len_max=-1)
@@ -138,7 +138,7 @@ class RNNGradTest(test.TestCase):
   def testBlockLSTMGradSeqLenMaxTooLarge(self):
     with self.assertRaisesRegex(
-        errors_impl.InvalidArgumentError, "seq_len_max must be <= timelen"
+        errors_impl.InvalidArgumentError, r"seq_len_max must be between 0 and"
     ):
       self.evaluate(self._block_lstm_grad(seq_len_max=10))
 
@@ -142,7 +142,7 @@ class RNNGradTest(test.TestCase):
   def testBlockLSTMGradSeqLenMaxNegative(self):
     with self.assertRaisesRegex(
-        errors_impl.InvalidArgumentError, "seq_len_max must be >= 0"
+        errors_impl.InvalidArgumentError, r"seq_len_max must be between 0 and"
     ):
       self.evaluate(self._block_lstm_grad(seq_len_max=-1))
```

**Comment by veerjain-1** [comment]
Hi @dmiltr3, thanks for investigating the CI failure! I have updated the expected error message regex in `rnn_grad_test.py` as suggested, and also fixed some PyLint line-length errors in the unit tests I added. Let me know if you need anything else!

**Review by dmiltr3 — APPROVED** [review]
All tests and verification checks have passed. Changes look good.

**Comment by veerjain-1** [comment]
I've applied the requested test fix for the regex. Could you please take another look and resolve the block if everything looks good? Also, I noticed some CI failures (`build-windows-x86` and `build-linux-x86-cpu`) - do those look like unrelated flaky tests, or is there something else I need to address on my end?

**Comment by nithyak0204** [comment]
Hi @veerjain-1, this PR is waiting for internal approvals. We are working on it. Thank you!
